packages/core/src/item.ts

```diff
@@ -32,8 +32,10 @@ export enum FieldType {
 export interface FieldDef {
     /** content type */
     type: FieldType;
-    /** regular expression describing pattern of field contents (not used for validation) */
+    /** regular expression describing pattern of field contents (used for validation) */
     pattern: RegExp;
+    /** regular expression describing pattern of field contents (used for matching) */
+    matchPattern: RegExp;
```

Comment:
I've created this other one that won't affect validation, and can be "stricter" (like forcing a `\n` to exist for notes) for matching.